shotover/src/server.rs

```diff
@@ -814,3 +805,130 @@ impl Shutdown {
         self.shutdown = true;
     }
 }
+
+/// Keeps track of all currently pending requests.
+/// This allows error responses to be generated if the connection needs to be terminated before the response comes back.
+enum PendingRequests {
```

Comment:
Would this be prone to a massive memory usage by a badly coded client?
One could even code a client in a way to DOS shotover this way. I would put a hard cap somewhere (might even be configurable).